flexget/plugins/input/npo_watchlist.py

```diff
@@ -77,47 +116,170 @@ def _get_profile(self, task, config):
 
         if profile_response.url == 'https://mijn.npo.nl/sessions':
             raise plugin.PluginError('Failed to login. Check username and password.')
-        elif profile_response.url != 'https://mijn.npo.nl/profiel':
-            raise plugin.PluginError('Unexpected profile page: {}'.format(profile_response.url))
+        elif profile_response.url != url:
+            raise plugin.PluginError('Unexpected page: {} (expected {})'.format(profile_response.url, url))
 
         return profile_response
 
-    def on_task_input(self, task, config):
+    def _get_watchlist_entries(self, task, config):
         email = config.get('email')
-        log.info('Retrieving npo.nl watchlist for %s', email)
+        log.info('Retrieving npo.nl episode watchlist for %s', email)
 
-        profile_response = self._get_profile(task, config)
-        profile_page = BeautifulSoup(profile_response.content, 'html5lib')
+        response = self._get_page(task, config, 'https://mijn.npo.nl/profiel/kijklijst')
+        page = BeautifulSoup(response.content, 'html5lib')
```

Comment:
You should use https://github.com/Flexget/Flexget/blob/develop/flexget/utils/soup.py#L15 instead